### PR TITLE
Merge processing pipelines

### PR DESCRIPTION
**Pull request by sandrinebedard**
# Context

This PR mainly:
* Updates README with new processing procedure.
* Merges processing of [preprocess_data.sh](https://github.com/sct-pipeline/contrast-agnostic-softseg-spinalcord/blob/main/preprocess_data.sh) and [process_data.sh](https://github.com/sct-pipeline/contrast-agnostic-softseg-spinalcord/blob/main/process_data.sh).
* Adds cropping and creation of `data_processed_clean` in the output folder.

Only the images that have segmentations and soft segmentations are transfered in `data_processed_clean `. 

## Procedure to test

1. Download the [spine-generic/data-multi-subject](https://github.com/spine-generic/data-multi-subject) by following the [procedure](https://github.com/spine-generic/data-multi-subject#download)
2. Go to the branch by running: `git checkout sb/121-move-derivatives`
3. Launch processing:

~~~
sct_run_batch -jobs -1 -path-data <PATH_DATA> -path-output <PATH-OUTPUT> -script process_data.sh -script-args exclude_reg.yml
~~~

An example of the results can be found here: `duke:projects/ivadomed/contrast-agnostic-seg/data-multi-subject-test13`

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -45,32 +27,56 @@ For DWI:
...
-#### Launch preprocessing
-Launch preprocessing on the [spine-generic multi-subject](https://github.com/spine-generic/data-multi-subject/releases/tag/r20220125) dataset:
-~~~
-sct_run_batch -jobs -1 -path-data <PATH-DATA> -path-output ~/sg_preprocess -script preprocess_data.sh
-~~~
-
-### Data processing
-
-These processed dataset then need to be input into [another processing scripts](https://github.com/sct-pipeline/contrast-agnostic-softseg-spinalcord/blob/main/process_data.sh) that:
+Next steps are to generate a constrast-agnostic soft segmentation:
```
Comment (jcohenadad) [inline README.md:30]:
```suggestion
Next steps are to generate a contrast-agnostic soft segmentation: 
```

**Review by jcohenadad — COMMENTED** [review]
not affected by this PR, but I realized this typo (constrat_for_seg --> contrast_for_seg):

https://github.com/sct-pipeline/contrast-agnostic-softseg-spinalcord/blob/68251b46e3a6b4d24d892174fab67204028cf871/process_data.sh#L92

https://github.com/sct-pipeline/contrast-agnostic-softseg-spinalcord/blob/68251b46e3a6b4d24d892174fab67204028cf871/process_data.sh#L112

File: README.md [hunk 2]
```diff
@@ -45,32 +27,56 @@ For DWI:
...
-### Data processing
-
-These processed dataset then need to be input into [another processing scripts](https://github.com/sct-pipeline/contrast-agnostic-softseg-spinalcord/blob/main/process_data.sh) that:
+Next steps are to generate a constrast-agnostic soft segmentation: 
 - creates spinal cord segmentation (if it doesn't exist yet)
 - creates a mask around the T2 spinal cord
 - co-register all contrasts to the T2 spinal cord 
 - average all segmentations from each contrast within the same space (the T2)
 - bring back the segmentations to the original image space of each contrast (except for the T2)
+- Crop images, segmentations and soft segmentation arround the spinal cord
```
Comment (jcohenadad) [inline README.md:36]:
```suggestion
- Crop images, segmentations and soft segmentation around the spinal cord
```

File: README.md [hunk 3]
```diff
@@ -45,32 +27,56 @@ For DWI:
...
 
 The output of this script is a new 'derivatives/labels_softseg/' folder that contains the soft labels to be used in this contrast-agnostic segmentation project. All the registration were manually QC-ed and the problematic registrations were listed in [exclude_reg.yml](https://github.com/sct-pipeline/contrast-agnostic-softseg-spinalcord/blob/main/exclude_reg.yml). The processing was run again to generate the soft segmentations.
```
Comment (jcohenadad) [inline README.md:38]:
```suggestion
The output of this script is a new `derivatives/labels_softseg/` folder that contains the soft labels to be used in this contrast-agnostic segmentation project. All the registration were manually QC-ed and the problematic registrations were listed in [exclude_reg.yml](https://github.com/sct-pipeline/contrast-agnostic-softseg-spinalcord/blob/main/exclude_reg.yml). The processing was run again to generate the soft segmentations. 
```

File: README.md [hunk 4]
```diff
@@ -45,32 +27,56 @@ For DWI:
...
+After running the analysis, check your Quality Control (qc) report by opening the file <PATH-OUTPUT>/qc/index.html. Use the "search" feature of the QC report to quickly jump to segmentations or labeling issues.
+
+**1. Segmentations**
+
+If segmentation issues are noticed while checking the quality report, proceed to manual correction using the procedure below:
+
+* In QC report, search for "deepseg" to only display results of spinal cord segmentation
+* Review segmentation and spinal cord.
+* Click on the F key to indicate if the segmentation is OK ✅, needs manual correction ❌ or if the data is not usable ⚠️ (artifact). Two .yml lists, one for manual corrections and one for the unusable data, will automatically be generated.
+* Download the lists by clicking on **Download QC Fails*** and on **Download Qc Artifacts**.
```
Comment (jcohenadad) [inline README.md:61]:
```suggestion
* Download the lists by clicking on **Download QC Fails*** and on **Download QC Artifacts**.
```

File: README.md [hunk 5]
```diff
@@ -45,32 +27,56 @@ For DWI:
...
+
+Proceed to manual correction using FSLeyes or ITK snap. Uplaod the manual segmentations (_seg-manual.nii.gz) with json sidecar in the derivatives.
+Re-run the analysis: [Launch processing](#launch-processing)
+
+**2. Registrations**
+
+* In QC report, search for "sct_register_multimodal" to only display results of registration.
+* Click on the F key to indicate if the registration is OK ✅, needs to be excluded ❌ or if the data is not usable ⚠️ (artifact). Two .yml lists, will automatically be generated.
+* Downlaod the list by clicking on **Download QC Fails** and add the file names under `FILES_REG` to https://github.com/sct-pipeline/contrast-agnostic-softseg-spinalcord/blob/main/exclude_reg.yml
```
Comment (jcohenadad) [inline README.md:70]:
```suggestion
* Download the list by clicking on **Download QC Fails** and add the file names under `FILES_REG` to https://github.com/sct-pipeline/contrast-agnostic-softseg-spinalcord/blob/main/exclude_reg.yml
```

File: README.md [hunk 6]
```diff
@@ -45,32 +27,56 @@ For DWI:
...
+* Re-run the analysis: [Launch processing](#launch-processing)
+
+**3. Soft segmentations**
+
+* In QC report, search for "softseg" to only display results of spinal cord soft segmentation
+* Click on the F key to indicate if the soft segmentation is OK ✅, needs manual correction ❌ or if the data is not usable ⚠️ (artifact). Two .yml lists, one for manual corrections and one for the unusable data, will automatically be generated.
+* Downlaod the list by clicking on **Download QC Fails**.
```
Comment (jcohenadad) [inline README.md:77]:
```suggestion
* Download the list by clicking on **Download QC Fails**.
```

File: README.md [hunk 7]
```diff
@@ -45,32 +27,56 @@ For DWI:
...
+* Uplaod the soft segmentations under `data-multi-subject/darivatives/labels_softseg`.
```
Comment (jcohenadad) [inline README.md:78]:
```suggestion
* Upload the soft segmentations under `data-multi-subject/derivatives/labels_softseg`.
```

write 100 times 'upload/download' on the black board 😉

Comment (sandrinebedard) [reply README.md]: it was more of a copy/paste error 😂

**Review by jcohenadad — APPROVED** [review]
I have not tested, but it looks good 😊

**Pull request merged**